pkg/sync/v8/data/ConfigMap/openshift-azure-logging/log-analytics-agent-entrypoint.yaml

```diff
@@ -0,0 +1,295 @@
+apiVersion: v1
+data:
+  main.sh: |-
+    #!/bin/bash
+
+    # NOTE: This script was lifted from /opt/main.sh in the azuremonitor/containerinsights/ciprod:ciprod07092019
+    # image and the docker group creation logic was modified by directly writing the group mapping to /etc/group.
+    # This script is to be mounted to the log analytics agent pods at /opt/main.sh, the same entrypoint configured
+    # for the image. This script should be removed as soon as we converge with upstream.
+
+    if [ -e "/etc/config/kube.conf" ]; then
+        cat /etc/config/kube.conf > /etc/opt/microsoft/omsagent/sysconf/omsagent.d/container.conf
+    else
+        sed -i -e 's/bind 127.0.0.1/bind 0.0.0.0/g' /etc/opt/microsoft/omsagent/sysconf/omsagent.d/container.conf
+    fi
+    sed -i -e 's/bind 127.0.0.1/bind 0.0.0.0/g' /etc/opt/microsoft/omsagent/sysconf/omsagent.d/syslog.conf
+    sed -i -e 's/^exit 101$/exit 0/g' /usr/sbin/policy-rc.d
+
+    #Using the get_hostname for hostname instead of the host field in syslog messages
+    sed -i.bak "s/record\[\"Host\"\] = hostname/record\[\"Host\"\] = OMS::Common.get_hostname/" /opt/microsoft/omsagent/plugin/filter_syslog.rb
+
+    #using /var/opt/microsoft/docker-cimprov/state instead of /var/opt/microsoft/omsagent/state since the latter gets deleted during onboarding
+    mkdir -p /var/opt/microsoft/docker-cimprov/state
+
+    #if [ ! -e "/etc/config/kube.conf" ]; then
+      # add permissions for omsagent user to access docker.sock
+      #sudo setfacl -m user:omsagent:rw /var/run/host/docker.sock
+    #fi
+
+    # add permissions for omsagent user to access azure.json.
+    sudo setfacl -m user:omsagent:r /etc/kubernetes/host/azure.json
+
+    # add permission for omsagent user to log folder. We also need 'x', else log rotation is failing. TODO: Investigate why.
+    sudo setfacl -m user:omsagent:rwx /var/opt/microsoft/docker-cimprov/log
+
+    DOCKER_SOCKET=/var/run/host/docker.sock
+    DOCKER_GROUP=docker
+    REGULAR_USER=omsagent
+
+    if [ -S ${DOCKER_SOCKET} ]; then
+        echo "getting gid for docker.sock"
+        DOCKER_GID=$(stat -c '%g' ${DOCKER_SOCKET})
+        echo "adding omsagent user to docker group ${DOCKER_GID}"
+        cat >>/etc/group <<EOL
+    ${DOCKER_GROUP}:x:${DOCKER_GID}:${REGULAR_USER}
+    EOL
+    fi
```

Comment:
This if block is the only place where main.sh is changed. Compare it to https://github.com/microsoft/OMS-docker/blob/2b721554eff9ed9108368f7ac959ece589e9f7e9/ci_feature_prod/main.sh#L32-L39

Reply:
here is a way to reduce the copying : https://github.com/openshift/openshift-azure/commit/cb4e803fd48b9ed8abaea3b6252695e1e0f0a692

Reply:
incorporated your suggestion